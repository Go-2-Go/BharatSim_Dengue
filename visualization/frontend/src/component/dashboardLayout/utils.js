import React from 'react';
import LineChart from '../lineChart/LineChart';
import BasicBarChart from '../barChart/BarChart';

const selectChart = (chartType, config) => {
  return {
    line: <LineChart config={config} />,
    bar: <BasicBarChart config={config} />,
  }[chartType];
};

export function createElement(element) {
  return (
    <div key={element.i} data-grid={element} data-testid={element.i}>
      {selectChart(element.chartType, element.config)}
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
