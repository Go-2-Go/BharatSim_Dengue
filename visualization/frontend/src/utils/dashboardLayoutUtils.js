/* eslint-disable react/prop-types */

import React from 'react';
import renderChart from '../component/charts/renderChart';

export function renderElement({ layout, chartType, config }) {
  return (
    <div key={layout.i} data-grid={layout} data-testid={layout.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(chartType, { config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
