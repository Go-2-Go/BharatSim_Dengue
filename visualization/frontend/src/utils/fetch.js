import axios from 'axios';
import { httpMethods } from '../constants/fetch';

let shouldShowError;

function commonErrorHandler(err) {
  shouldShowError(true);
}

function initApiConfig({ setError }) {
  shouldShowError = setError;
}

function fetchData({ url, method = httpMethods.GET, headers, data, query, customErrorHandler }) {
  return axios({ url, method, headers, data, params: query })
    .then((res) => res.data)
    .catch((err) => {
      if (customErrorHandler) {
        customErrorHandler(err);
        return;
      }
      commonErrorHandler(err);
    });
}

function uploadData({ url, headers, data, query, customErrorHandler }) {
  return fetchData({
    url,
    method: httpMethods.POST,
    headers,
    data,
    params: query,
    customErrorHandler,
  });
}

export { fetchData, uploadData, initApiConfig };
