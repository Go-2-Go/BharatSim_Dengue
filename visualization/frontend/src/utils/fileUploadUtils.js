const fileUploadedStatus = {
  LOADING: 'loading',
  SUCCESS: 'success',
  ERROR: 'error',
};

function getStatusAndMessageFor(fileUploadStatus, fileName) {
  return {
    [fileUploadedStatus.ERROR]: {
      status: fileUploadedStatus.ERROR,
      message: `Error occurred while unloading ${fileName}`,
    },
    [fileUploadedStatus.SUCCESS]: {
      status: fileUploadedStatus.SUCCESS,
      message: `${fileName} successfully uploaded`,
    },
    [fileUploadedStatus.LOADING]: {
      status: fileUploadedStatus.LOADING,
      message: `uploading ${fileName}`,
    },
  }[fileUploadStatus];
}

export { fileUploadedStatus, getStatusAndMessageFor };
