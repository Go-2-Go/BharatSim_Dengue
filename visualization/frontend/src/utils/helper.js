function updateState(prevState,updatedValue){
  return Object.assign({},prevState,updatedValue)
}

export {
  updateState,
}