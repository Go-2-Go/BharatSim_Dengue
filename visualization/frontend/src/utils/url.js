const DATASOURCE_URL = '/api/dataSources';
const DASHBOARD_URL = '/api/dashboard';

export const url = {
  getDataUrl: (dataSource) => `${DATASOURCE_URL}/${dataSource}`,
  getHeaderUrl: (dataSource) => `${DATASOURCE_URL}/${dataSource}/headers`,
  DATA_SOURCES: `${DATASOURCE_URL}`,
  saveDashboard: DASHBOARD_URL,
};
