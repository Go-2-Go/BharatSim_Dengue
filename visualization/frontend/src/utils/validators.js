const isAbsent = (value) => {
  return value === null || value === undefined || value === '';
};

function isNotAxisTypeNumber(value) {
  return value.type !== 'number';
}

function isEmptyObject(value) {
  return Object.keys(value).length === 0;
}

const xAxisValidator = (value = '') => {
  if (isAbsent(value)) {
    return 'Please select value for x axis';
  }
  return '';
};

const yAxisValidator = (value = {}) => {
  if (isAbsent(value)) {
    return 'Please select value for y axis';
  }
  if (isEmptyObject(value)) {
    return 'Please select value for y axis';
  }
  if (isNotAxisTypeNumber(value)) {
    return 'Please select number type option';
  }
  return '';
};

const datasourceValidator = (value = '') => {
  if (isAbsent(value)) {
    return 'Please select data source';
  }
  return '';
};

export { datasourceValidator, xAxisValidator, yAxisValidator };
