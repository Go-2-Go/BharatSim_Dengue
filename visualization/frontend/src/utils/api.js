import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async () => fetchData({ url: serviceURL.DATA_SOURCES }),

  getData: async (datasource, query) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
