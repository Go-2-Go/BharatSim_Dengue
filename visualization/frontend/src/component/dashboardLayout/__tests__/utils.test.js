import React from "react";
import {render} from '@testing-library/react'
import {createElement, getInitialLayout} from "../utils";

jest.mock("../../lineChart/LineChart.js", () => () => {
  return <>Hello, Welcome</>
})

describe('Dashboard layout utils', function () {
  describe("Create element", () => {
    it('should provide element with data-grid', () => {
      const element = createElement({i: 'id-1'});
      const {container} = render(<>{element}</>);

      expect(container).toMatchSnapshot();
    })
  })

  describe("getInitialLayout", () => {
    it('should provide element with data-grid', () => {
      const layout = getInitialLayout();

      expect(layout).toEqual([{
        i: '0',
        x: 0,
        y: 0,
        w: 2,
        h: 2,
      }]);
    })
  })
});