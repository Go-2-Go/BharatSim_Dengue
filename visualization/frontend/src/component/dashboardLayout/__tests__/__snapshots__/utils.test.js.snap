// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard layout utils Create element should provide element with data-grid 1`] = `
<div>
  <div
    data-grid="[object Object]"
    data-testid="id-1"
  />
</div>
`;
